vizro-core/src/vizro/static/js/models/page.js

```diff
@@ -156,44 +155,22 @@ function sync_url_query_params_and_controls(...values_ids) {
     `${window.location.pathname}?${urlParams.toString()}`,
   );
 
-// // Two approaches:
-// // 1. Don't return filter values but use setProps instead. ---> Does NOT work as setProps trigger filter_action.
-// // 2. Return SELECTOR_guard_actions_chain from the callback output too.
-//
-// // 1:
-//  const outputSelectorIds = dash_clientside.callback_context.outputs_list;
-//  for (let i = 1; i < outputSelectorIds.length; i++) {
-//    const selectorId = outputSelectorIds[i]['id'];
-//    const selectorValue = outputSelectorValues[i - 1];
-//
-//
-//    if (selectorValue !== dash_clientside.no_update) {
-//      dash_clientside.set_props(selectorId, {value: selectorValue});
-//    }
-//  }
-//
-//  return [triggerOPL, ...outputSelectorValuesAllNoUpdate];
-
-  // 2:
-  const outputSelectorIds = dash_clientside.callback_context.outputs_list;
-  for (let i = 1; i < outputSelectorIds.length; i++) {
-    const selectorId = outputSelectorIds[i]['id'];
-    const selectorValue = outputSelectorValues[i - 1];
-
+  // After this clientside callback, the "guard_action_chain" callback may run.
+  // If the selector value is updated based on the URL parameters,
+  // set its values and the selector’s guard flag to **true**.
+  // This ensures triggering the guard action chain callback
+  // and prevents unnecessary actions from being triggered by the value change.
+  for (let i = 0; i < selectorIds.length; i++) {
+    const selectorId = selectorIds[i];
+    const selectorValue = outputSelectorValues[i];
```

Comment:
This is probably slightly more idiomatic javascript:

```suggestion
selectorIds.forEach((selectorId, i) => {
  const selectorValue = outputSelectorValues[i];
```